spec/classes/init_spec.rb

```diff
@@ -2335,6 +2335,50 @@
     end
   end
 
+  describe 'with ssh_key_export parameter specified' do
+    context 'as a non-boolean or non-string' do
+    let(:params) { { :ssh_key_export => ['not_a_boolean','or_a_string'] } }
+
+      it 'should fail' do
+        expect {
+          should contain_class('ssh')
+        }.to raise_error(Puppet::Error)
+      end
+    end
+
+    context 'as an invalid string' do
+      let(:params) { { :ssh_key_export => 'invalid_string' } }
+
+      it 'should fail' do
+        expect {
+          should contain_class('ssh')
+        }.to raise_error(Puppet::Error,/ssh::ssh_key_export may be either 'true' or 'false' and is set to <invalid_string>\./)
+      end
+    end
+
+    ['true',true].each do |value|
+      context "as #{value}" do
+        let(:params) { { :ssh_key_export => value } }
+
+        it { should compile.with_all_deps }
+
+        it { should contain_class('ssh') }
+
+        it { should contain_resources('sshkey') }
+      end
+    end
+
+    ['false',false].each do |value|
+      context "as #{value}" do
+        let(:params) { { :ssh_key_export => value } }
+
+        it { should compile.with_all_deps }
+
+        it { should contain_class('ssh') }
```

Comment:
When false, it should check that the sshkey is not present.

```ruby
it { should_not contain_resources('sshkey') }
```

Reply:
I already tried that but the resource is still in the catalog, just not exported.

Reply:
Should be done.